.github/workflows/gradle-compiler.yml

```diff
@@ -34,6 +34,16 @@ jobs:
         java-version: '17'
         java-package: jdk
         architecture: x64
+        cache: gradle
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3
+
+    - name: Validate Gradle Wrapper
+      uses: gradle/actions/wrapper-validation@v3
+
+    - name: Grant Wrapper Permission
```

Comment:
I suggest we do this locally once (and push) instead of doing it every time in every task we want to invoke gradle

Reply:
See also https://stackoverflow.com/questions/40978921/how-to-add-chmod-permissions-to-file-in-git